platform/default/include/mbgl/util/default_thread_pool.hpp

```diff
@@ -6,17 +6,40 @@
 #include <mutex>
 #include <queue>
 #include <thread>
+#include <functional>
 
 namespace mbgl {
 
+struct ThreadLifecycle {
```

Comment:
isn't It and observer? Maybe call it `ThreadObserver` and equip with two virtual methods: `onThreadCreated` and `onThreadDestroyed`. Further, you could override it for Android and get rid of std::functions.

Reply:
:smile: that was initial patch from @LukasPaczos, there were few problems:

- cannot be implemented by `AndroidThreadPool`, as we would need to call virtual methods from constructor or split constructor and starting of the threads in thread pool.
- if implemented by another object, or even by `AndroidThreadPool`, observer methods would not be that useful, as there is a need to pass thread specific data.

That is the reason why I proposed to introduce simple copyable object that is used on the started thread.